scripts/validate_docstrings.py

```diff
@@ -331,6 +336,33 @@ def parameter_mismatches(self):
 
         return errs
 
+    @contextlib.contextmanager
+    def _write_examples_code_to_temp_file(self):
+        """
+        Generate file with source code from examples section.
+        """
+        content = ''.join(('import numpy as np; '
+                           'import pandas as pd  # noqa: F401,E702\n',
+                           *self.examples_source_code))
```

Comment:
wouldn't it be simpler to join by `\n` instead, and remove the semicolon and the noqa?

Reply:
* regarding E702: with one line the great thing was to achieve intercepting of the error message.
* regarding F401: All examples not using both: np and pd would fail because of an unused import.

The implementation of `self.examples_source_code` of PR #23161 returns an array with `\n` at the end, so joining with `\n` isn't possible.